react-charts/bar-chart/step4.md

```diff
@@ -1,65 +1,52 @@
 
-1) Make sure the App.js file is still open
-
-2) Locate the code that looks like the following:
+2) <strong>Locate the `<Chart>` component.</strong>
 
 <pre class="file">
-&lt;ChartBar 
-  data={[
-    { name: &#39;Cats&#39;, x: &#39;2015&#39;, y: 1 }, 
-    { name: &#39;Cats&#39;, x: &#39;2016&#39;, y: 2 }, 
-    { name: &#39;Cats&#39;, x: &#39;2017&#39;, y: 5 }, 
-    { name: &#39;Cats&#39;, x: &#39;2018&#39;, y: 3 }
-  ]} 
-/&gt;
+&lt;Chart
+  domainPadding={{ x: [30, 25] }}
+  height={250}
+  width={600}
+&gt;
 </pre>
 
-3) Replace that section with the following code:
+3) <strong>Add a `containerComponent` property to the `Chart` component.</strong>
+
+The `containerComponent` property takes a component instance, which will be used
+to create a container element for standalone charts.
+
+Inside the `containerComponent` is the `ChartVoronoiContainer` component, which
+adds the ability to associate a mouse position with the data point(s) closest to
+it. This is useful for adding a tooltip.
+
+It should look like this:
 
 <pre class="file" data-target="clipboard">
-&lt;ChartGroup offset={11}&gt;
-  &lt;ChartBar
-    data={[
-      { name: &#39;Cats&#39;, x: &#39;2015&#39;, y: 1 }, 
-      { name: &#39;Cats&#39;, x: &#39;2016&#39;, y: 2 }, 
-      { name: &#39;Cats&#39;, x: &#39;2017&#39;, y: 5 }, 
-      { name: &#39;Cats&#39;, x: &#39;2018&#39;, y: 3 }
-    ]} 
-  /&gt;
-  &lt;ChartBar 
-    data={[
-      { name: &#39;Dogs&#39;, x: &#39;2015&#39;, y: 2 }, 
-      { name: &#39;Dogs&#39;, x: &#39;2016&#39;, y: 1 }, 
-      { name: &#39;Dogs&#39;, x: &#39;2017&#39;, y: 7 }, 
-      { name: &#39;Dogs&#39;, x: &#39;2018&#39;, y: 4 }
-    ]}
-  /&gt;
-  &lt;ChartBar 
-    data={[
-      { name: &#39;Birds&#39;, x: &#39;2015&#39;, y: 4 }, 
-      { name: &#39;Birds&#39;, x: &#39;2016&#39;, y: 4 }, 
-      { name: &#39;Birds&#39;, x: &#39;2017&#39;, y: 9 }, 
-      { name: &#39;Birds&#39;, x: &#39;2018&#39;, y: 7 }
-    ]}
-  /&gt;
-  &lt;ChartBar 
-    data={[
-      { name: &#39;Mice&#39;, x: &#39;2015&#39;, y: 3 }, 
-      { name: &#39;Mice&#39;, x: &#39;2016&#39;, y: 3 }, 
-      { name: &#39;Mice&#39;, x: &#39;2017&#39;, y: 8 }, 
-      { name: &#39;Mice&#39;, x: &#39;2018&#39;, y: 5 }
-    ]}
+containerComponent={
+  &lt;ChartVoronoiContainer
   /&gt;
-&lt;/ChartGroup&gt;
+}
 </pre>
 
-- The `ChartGroup` is the React Component used to apply properties to multiple `ChartBar` components
-- The `offset` property is used to adjust each `ChartBar` from its original position
+<strong>Note:</strong> There are no visible changes at this stage.
+
+4) <strong>Add a `constrainToVisibleArea` property inside of the `ChartVoronoiContainer`.</strong>
+
+The `constrainToVisibleArea` property determines whether to coerce tooltips so that they fit within the visible area of the chart.
+
+It should look like this:
+
+<pre class="file" data-target="clipboard">
+containerComponent={
+  &lt;ChartVoronoiContainer
+    constrainToVisibleArea
+  /&gt;
+}
+</pre>
 

```

Comment:
Need a fifth step here for adding the labels: `labels={({ datum }) => `${datum.name}: ${datum.y}`}`, look at how Area-charts does it. this step is broken otherwise